helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1165,7 +1163,11 @@ public void run() {
               + _processorName);
       while (!isInterrupted()) {
         try {
-          handleEvent(_eventBlockingQueue.take(), _cache);
+          ClusterEvent newClusterEvent = _eventBlockingQueue.take();
```

Comment:
Would it be better to add some helper methods within ClusterEvent that stores a "threadName", and modify by using a setter?

This way, we save an object creation.

Reply:
Don't you think it's too much for the simplicity of code? And logically, ClusterEvent shouldn't know the existence of threadName even queue object

Reply:
I don't think it's too much for the simplicity of the code, but I do think that it will be more work for you to make that happen.

I don't know if I follow "And logically, ClusterEvent shouldn't know the existence of threadName even queue object".

At any rate, the point still stands - it is always important to minimize the amount of objects being created since we need to be conscious of memory pressure. If you have other thoughts, please post them here.

I am fine with leaving this item as a **TODO** in the codebase and creating a corresponding issue.